File: Lecture031 Recursion Day1/factorial.py
def factorial(n):
    # base case
    if n == 0:
        return 1
    smallerProblem = factorial(n-1)
    biggerProblem = n * smallerProblem

    return biggerProblem

if __name__ == "__main__":
    n=int(input())
    ans = factorial(n)
    print(ans)

File: Lecture032 Recursion Day2/Factorial.java
import java.util.Scanner;

public class Factorial {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in); // Scanner to read values from user

        int n = sc.nextInt(); // take input from user

        int ans = fact(n); // calling fact function to compute our factorial value of n

        System.out.println(ans); // print the ans
    }

    public static int fact(int n){ // function to find the factorial
        //Base case
        if(n == 0)
            return 1;

        // Recurrence relation
        int smallerProblem = fact(n - 1);
        int biggerProblem = n * smallerProblem;

        //  can be also written as
        //  biggerProblem = n * fact(n - 1);

        return biggerProblem;
    }
}
